service/storage_service.cc

```diff
@@ -118,14 +118,15 @@ distributed<storage_service> _the_storage_service;
 
 
 timeout_config make_timeout_config(const db::config& cfg) {
-    timeout_config tc;
-    tc.read_timeout = cfg.read_request_timeout_in_ms() * 1ms;
-    tc.write_timeout = cfg.write_request_timeout_in_ms() * 1ms;
-    tc.range_read_timeout = cfg.range_request_timeout_in_ms() * 1ms;
-    tc.counter_write_timeout = cfg.counter_write_request_timeout_in_ms() * 1ms;
-    tc.truncate_timeout = cfg.truncate_request_timeout_in_ms() * 1ms;
-    tc.cas_timeout = cfg.cas_contention_timeout_in_ms() * 1ms;
-    tc.other_timeout = cfg.request_timeout_in_ms() * 1ms;
+    timeout_config tc{
+        .read_timeout = utils::updateable_value<db::timeout_clock::duration>(cfg.read_request_timeout_in_ms() * 1ms),
+        .write_timeout = utils::updateable_value<db::timeout_clock::duration>(cfg.write_request_timeout_in_ms() * 1ms),
```

Comment:
Designated initializers are not in C++17 (they're a gcc extension), let's only use them with C++20.


How is this updateable? by doing the multipication, you broke the link with the updateable_value_source.

Reply:
I'll remove the designated initialization, I tried to keep the code as close as possible to the original and was happy that it compiled for me...

I'll have  second look to be sure or update the implementation